R/postprocess.R

```diff
@@ -1,3 +1,132 @@
+#' Extract samples of the delay distribution parameters
+#'
+#' @param fit A model fit with `epidist::epidist`
+#' @param ... Additional arguments passed to `brms::prepare_predictions`
+#' @family postprocess
+#' @autoglobal
+#' @export
+delay_samples <- function(fit, ...) {
+  args <- list(...)
+  if ("newdata" %in% names(args)) {
+    newdata <- brms:::validate_newdata(newdata)
```

Comment:
I think we might want two user functions rather than overloading what we have? 

One (i.e this one) does extraction of draws based on what was fitted and the other (i.e `predict_delays` etc.) handles having a new data input. Both can produce an object of the same class though and share downstream functionality? What do you think?

Reply:
Alternatively I think we should follow more closely to the standard `predict` or `posterior_predict` naming scheme and always have new data as an argument.

Reply:
If a user wants to extract draws of parameters from the model they can use standard `brms` and Stan functionality.

These functions are to extract useful functions of parameters.

I guess to be concrete: what do you mean by "extraction of draws based on what was fitted". Is this the same as `delay_samples` with `newdata = NULL` which will get draws for each data point (individual).

About including `newdata` within the names or not, this was something I was unsure about. I'm happy to include it. It is included in `...` but perhaps it's hard for users to know that. The consideration in my head was like "making it obvious by specially pointing to `newdata`" versus "treating all other arguments of `brms::prepare_predictions` as on an equal footing to `newdata`": who knows maybe users want to use those too? But yes I do think `newdata` can have a special place.

Reply:
Talked through names F2F and arrived at `predict_delay_parameters` (maybe with alias for `predict_dpars`)